### ISSUE TITLE
bug: gwc-button-select No change event on Safari iPadOs  

### ISSUE THREAD

**Issue by reno1979** [maritieme-it/gwc#23]
[case]
A gwc-button-select component from which the options are created to the property `items`.
When an option is clicked, the change event is not thrown when using Safari on an iPad.

[cause] 
The click on the gwc-button-select-option element bubbles up from the shadowDom.
The onclick handler from the gwc-button-select captures this event and uses the path it finds in the click event to determine if the event is caused by a gwc-button-select-option. 

Two things wrong here: 
`event.path` is undefined, instead of `path` it is adviced to used event.composedPath();
The event.composedPath() gives an empty array inside Safari on tap event. 

[solution]
Let gwc-button-select-option capture its own click and prevent it from travelling any further.
let gwc-button-select-option throw its own event. (name `gwc-button-select-option-click` ? )
let gwc-button-select capture the custom event from the gwc-button-select-option 
etc

**Comment by krispijn** [comment]
> The second solution would be to:
Let gwc-button-select-option capture its own click and prevent it from travelling any further.
let gwc-button-select-option throw its own event. (name gwc-button-select-option-click ? )
let gwc-button-select capture the custom event from the gwc-button-select-option
etc

Wasn't this already implemented this way in the old setup (with separate gwc-button-select-option folder)?

**Comment by reno1979** [comment]
> > The second solution would be to:
> > Let gwc-button-select-option capture its own click and prevent it from travelling any further.
> > let gwc-button-select-option throw its own event. (name gwc-button-select-option-click ? )
> > let gwc-button-select capture the custom event from the gwc-button-select-option
> > etc
> 
> Wasn't this already implemented this way in the old setup (with separate gwc-button-select-option folder)?

Correct, the options acted as buttons, each with its own click listener. On click it would toggle its own active state and dispatch a custom event. Which was captured by the gwc-button-select.

```
this.addEventListener(gwcButtonSelectOption.NAME, this._handleOptionClick.bind(this));
```

On this custom event capturem the internal values got updated. 

```
_updateSelectedValues(event){
        if(this.multiselect){
            let _itOptions = this._retrieveAssignedOptionNodes(); 
            this._value             = []; 
            _itOptions.forEach(_elOption => {
                if(_elOption.active){
                    this._value.push(_elOption.value);
                }
            });
        } else {
            let _elOptionClicked = event.composedPath()[0];
            this._value      = event.detail; 
            this._resetSelectedOptions(); 
            _elOptionClicked.active = true;  // restore clicked element
            this.active         = false; // close option wrapper
        }
    }
```

This also used the composed path .... without validation .... oops. 

Also note that there is a difference between events captured on option elements assigned through slots and elements generated directly in the shadow DOM ( data/ items) property, and that this should be tested on on an iPad. 

When using the custom event solution we are creating an event listenerer for each option (in each button select), not the solution I would prefer. But ok, it will `work`.

The solution is not too hard, I posted this bug because of the empty composedPath() on the iPad on touch events (something gets lost in translation I guess)  And I expect this to cross our path once more in the near future. (click events on children created inside the shadowdom show the origin as being the shadowDom (event.target etc ) )
